### PR TITLE
Remove ez_setup from documentation

### LINKED ISSUE #1540: Developer's Guide should not refer to ez_setup.py
While attempting to figure out how to run the test suite, I ran "manual_test.py" in the test directory. This threw an assertion error saying I needed to have setuptools installed from the "dev" distribution.

Documentation here refers to using ez_setup.py to install dev version

https://setuptools.readthedocs.io/en/latest/setuptools.html#installing-setuptools

### PR DESCRIPTION
**Pull request by varunkamath18**
## Summary of changes

Split up previous "Installing setuptools" section into the following:
1. Use link to pip installation instead of Easy install instructions 
2. Move "Custom installation instructions" to documentation for creating virtual environments.

Closes #1540

### Pull Request Checklist
- [ ] Changes have tests
- [x] News fragment added in changelog.d. See [documentation](http://setuptools.readthedocs.io/en/latest/developer-guide.html#making-a-pull-request) for details

### CODE REVIEW

File: docs/setuptools.txt [hunk 1]
```diff
@@ -73,23 +73,22 @@ Developer's Guide
...
-    ez_setup.py setuptools==dev
+.
 
-This will download and install the latest development (i.e. unstable) version
-of setuptools from the Python Subversion sandbox.
+In particular, be sure to read the
+section on `Creating Virtual Environments`_ if you're interested in installing 
+anywhere other than  ``site-packages`` directory.
 
+.. _Creating Virtual Environments: https://packaging.python.org/tutorials/installing-packages/#creating-virtual-environments
```
Comment (benoit-pierre) [inline docs/setuptools.txt]: I don't think we need to mention the virtualenv thing, I'd just use something along those lines:
```
To install setuptools, use:

    pip install setuptools

Refer to the `Installing Packages`_ guide for more information.
```
My problem with those instructions is they don't use `--user` by default, and we really don't want users to use `sudo pip` to work around that...

Comment (varunkamath18) [reply docs/setuptools.txt]: Thanks, will make that change now.

File: docs/setuptools.txt [hunk 2]
```diff
@@ -73,23 +73,17 @@ Developer's Guide
...
 
-.. _EasyInstall Installation Instructions: easy_install.html#installation-instructions
+.. _Custom Installation Locations: easy_install.html
 
-.. _Custom Installation Locations: easy_install.html#custom-installation-locations
+.. _Installing Packages: https://packaging.python.org/tutorials/installing-packages/
 
-If you want the current in-development version of setuptools, you should first
-install a stable version, and then run::
+To install setuptools, use::
```
Comment (pganssle) [inline docs/setuptools.txt]:
```suggestion
To install the latest version of setuptools, use::
```

File: docs/setuptools.txt [hunk 3]
```diff
@@ -73,23 +73,17 @@ Developer's Guide
...
 
-    ez_setup.py setuptools==dev
-
-This will download and install the latest development (i.e. unstable) version
-of setuptools from the Python Subversion sandbox.
+    pip install setuptools
```
Comment (pganssle) [inline docs/setuptools.txt]:
```suggestion
    pip install -U setuptools
```

**Comment by pganssle** [comment]
Oh actually, also this needs a changelog.

**Pull request merged**